pkg/netpol/eval/resources.go

```diff
@@ -352,11 +344,16 @@ func (pe *PolicyEngine) upsertNetworkPolicy(np *netv1.NetworkPolicy) error {
 	}
 	pe.netpolsMap[netpolNamespace][np.Name] = newNetpol
 
-	// for exposure analysis only: scan policy ingress and egress rules to store allowed connections
-	// to entire cluster and to all destinations (if such connections are allowed by the policy)
 	var err error
+	// for exposure analysis only: scan policy ingress and egress rules:
+	// 1. to store allowed connections to entire cluster and to all destinations (if such connections are allowed by the policy)
+	// 2. to get selectors and generate representativePeers (each specified rule, gets a representative peer)
 	if pe.exposureAnalysisFlag {
-		err = newNetpol.DetermineGeneralConnectionsOfPolicy()
+		rulesSelectors, scanErr := newNetpol.ScanPolicyRulesForGeneralConnsAndRepresentativePeers()
+		if scanErr != nil {
+			return scanErr
+		}
+		err = pe.generateRepresentativePeers(rulesSelectors, np.Name)
```

Comment:
how about consider to sort the objects to add, so that netpols are added first, and then when workloads are add - for each added workload we can remove a redundant representative peer?
this may simplify the code, and let us avoid having redundant peers through the entire flow?

Reply:
actually I can sort only in case of exposure-analysis

Reply:
ok, so let's try to simplify.
for any added workload you will have to iterate the remaining representative peers and remove those which are of a matching selector, right?

Reply:
done